src/Driver.cpp

```diff
@@ -0,0 +1,91 @@
+#include "Driver.hpp"
+#include "Oculus.h"
+#include <string.h>
+
+using namespace sonar_oculus_m750d;
+
+Driver::Driver()
+    : iodrivers_base::Driver::Driver(INTERNAL_BUFFER_SIZE)
+{
+    m_protocol = Protocol();
+}
+
+bool isValidHeader(OculusMessageHeader const& header)
+{
+    return header.oculusId == OCULUS_CHECK_ID;
+}
+
+int Driver::extractPacket(uint8_t const* buffer, size_t buffer_size) const
+{
+    auto header_size = sizeof(OculusMessageHeader);
+
+    if (buffer_size < header_size) {
+        // Not enough data for a header — wait for more
+        return 0;
+    }
+
+    OculusMessageHeader header;
+    memcpy(&header, buffer, header_size);
+    if (!isValidHeader(header)) {
+        // discard first byte
+        return -1;
+    }
+
+    size_t packet_size = header_size + header.payloadSize;
+    if (buffer_size >= packet_size) {
```

Comment:
```suggestion
    if (buffer_size >= packet_size + payload_size) {
```

get the payload size in the header

Reply:
Unless I'm mistaken you'ved one this ? Please remove that comment ?